### DISCUSSION TITLE
MQTT integration and HomeAssistant

### DISCUSSION THREAD

**Discussion by mfoo** [martomi/chiadog#160] [Ideas]
Hi! I've just come across this project and I'm interested in checking it out this weekend. In particular, I use Home Assistant for home automation and some level of basic monitoring. As part of that I have an Apache Mosquitto MQTT instance that is used for some systems, and Home Assistant supports an `expire_after` key that you can use in MQTT messages after which it reports devices as unavailable and you can trigger automations based off of that.

From a look at the code, it'd be quite simple to add to the payload in the MQTT notification file.

https://www.home-assistant.io/integrations/sensor.mqtt/#expire_after

I'm not sure this is a normal thing to do with MQTT payloads though and I don't know what others are using it for. Would it be possible to add as a configuration setting?

Thanks!

**Comment by martomi** [comment]
@pieterhelsen might be able to provide feedback on how this fits the current use-cases of the MQTT notifier.

**Comment by pieterhelsen** [comment]
`expire_after` (like MQTT Auto Discovery) is solely used by Home Assistant. 
What you could do typically is write a little consumer script that subscribes to the chia topic, augments the JSON message and then publishes to a different topic which you would have Home Assistant pick up.
